conftest.py

```diff
@@ -88,3 +88,11 @@ def pytest_runtest_setup(item):
 def pytest_runtest_teardown(item, nextitem):
     if isinstance(item, DoctestItem):
         set_config(print_changed_only=False)
+
+
+# We don't want pytest to run these files since they immediately raise a
+# DeprecationWarning
+collect_ignore_glob = [
+    "sklearn/neural_network/rbm.py",  # 0.24
```

Comment:
does this also result in the doctests not to run?

Reply:
Yes, maybe rather not raise the warning in a pytest session?

Something like in the affected files,
```py
import sys
if not getattr(sys, '_is_pytest_session', False):
    warnings.warn(...)
```
where `sys._is_pytest_session` is already set in `conftest.py`

Reply:
There is no docstring in `rbm.py`. There's almost no code in these files now, all they do is raises a warning.

@rth I have no strong opinion but I think that we should rather have pytest ignore a file, rather than have a file have specific behavior for pytest. LMK.

Reply:
> There is no docstring in rbm.py. There's almost no code in these files now, all they do is raises a warning.

For now. Hard to to know some that this file is ignored by pytest just by looking at it (or to remove that line once the file is removed). I think if we can do deprecations locally without touching global config files it is better.

Reply:
> For now

I don't see the file evolving, ever? This is the file we just created.

We only need to ignore the file because we run the tests with `-Werror::DeprecationWarning`. Not raising the warning when we are in a pytest session is treating the issue in the reverse order IMO, but OK. I'll address it in #14939 	if you don't mind

Reply:
> I don't see the file evolving, ever? This is the file we just created.

Me neither, but who knows. Accidents and messed up merge conflicts do happen. 

> Not raising the warning when we are in a pytest session is treating the issue in the reverse order IMO, but OK.

Yes, I agree it's not ideal, but both this and changing the `conftest.py` is a way around our non standard use of error on warning. Not sure if there is a better solution.